### PR TITLE
Changed submodule revision to nanomsg 0.5beta tag + related changes

### PR DESCRIPTION
**Pull request by comick**
Modified nanomsg submodule revision to match 0.5beta tag. Some changes have been necessary:
- symbols.js regenerated to make tests successful
- files list in bynging.gyp synced and resorted matching nanomsg build order
All tests green.

Cheers,
M.

### CODE REVIEW

File: test/symbols.js [hunk 1]
```diff
@@ -76,6 +76,14 @@ exports.symbols = [
 
 },
 {
+                'value': nn.NN_NS_OPTION_UNIT,
+                'name':  "NN_NS_OPTION_UNIT",
+                'ns':    nn.NN_NS_NAMESPACE,
+                'type':  nn.			NN_TYPE_NONE,
```
Comment (nickdesaulniers) [inline test/symbols.js]: Why the whitespace?

**Comment by comick** [comment]
File was regenerated through mksymbols.rb. I didn't notice the whitespaces, which is actually 3 tabs.
Have to do with the input source in symbols.c:
```c
//...
    {NN_NS_OPTION_TYPE, "NN_NS_OPTION_TYPE", NN_NS_NAMESPACE,
        NN_TYPE_NONE, NN_UNIT_NONE},
     {NN_NS_OPTION_UNIT, "NN_NS_OPTION_UNIT", NN_NS_NAMESPACE,
            NN_TYPE_NONE, NN_UNIT_NONE},
    {NN_NS_FLAG, "NN_NS_FLAG", NN_NS_NAMESPACE,
        NN_TYPE_NONE, NN_UNIT_NONE},
//...
```

Anyway doesn't seem to be a problem. Maybe the generator can be modified to cleanup such intentation garbage.

**Comment by nickdesaulniers** [comment]
Please strip whitespace from the rest of the tokens in mksymbols.rb, like [here](https://github.com/nickdesaulniers/node-nanomsg/blob/c44ad6cd76b290bd675fc14a2091e0d41844ab7d/test/mksymbols.rb#L25) and see if that does the trick.

**Comment by nickdesaulniers** [comment]
> files list in bynging.gyp synced and resorted matching nanomsg build order

Also, I don't understand by what you mean by this?  Why is this necessary?  It makes it difficult to tell what new symbols were added, if any were removed, or if they were just sorted differently.

File: .gitignore [hunk 2]
```diff
@@ -15,3 +15,5 @@ nanocat
 perf
 .libs
 Makefile
+.idea
+*.iml
```
Comment (nickdesaulniers) [inline .gitignore]: What are .idea and .iml files?  Maybe a better place for these is `~/.gitignore_global`?

**Comment by comick** [comment]
Now strips white symbols.
Removed non-project .gitignore entries.
For the nanomsg files list, just got them from the output of nanomsg make and put in bynding.gyp accordingly once json-formatted. That's why the order.

**Comment by nickdesaulniers** [comment]
looks good!  Thanks!

**Comment by nickdesaulniers** [comment]
published in v0.3.2

**Comment by comick** [comment]
cool!

**Pull request merged**